doc/sphinx/language/core/modules.rst

```diff
@@ -511,6 +511,249 @@ transparent constraint
+   persist when the section is closed and applies the effect when the
+   module containing the command is imported.
+
+   Commands supporting this attribute include :cmd:`Set`, :cmd:`Unset`
+   and the :ref:`Hint <creating_hints>` commands, although the latter
+   don't support it within sections.
+
+.. attr:: global
+
+   This :term:`attribute` makes the effect of the command
+   persist even when the current section or module is closed.  Loading
+   the file containing the command (possibly transitively) applies the
+   effect of the command.
+
+   The ``Global`` prefix is an alternative syntax for the
+   :attr:`global` attribute (see :n:`@legacy_attr`).
+
+   .. warning::
+
+      **Exception:** for a few commands (like :cmd:`Notation` and
+      :cmd:`Ltac`), this attribute behaves like :attr:`export`.
+
+   .. warning::
+
+      We strongly discourage using the :attr:`global` locality
+      attribute because the transitive nature of file loading gives
+      the user little control. We recommend using the :attr:`export`
+      locality attribute where it is supported.
+
+.. _visibility-attributes-modules:
+
+Summary of locality attributes in a module
+------------------------------------------
+
+The following table sums up the locality of vernacular commands in modules, when
+outside the module where it was entered.
+A similar table for :cmd:`Section` can be found
+:ref:`here<visibility-attributes-sections>`.
+
+.. list-table::
+  :header-rows: 1
+
+  * - ``Command``
+    - without attribute
+    - :attr:`local`
+    - :attr:`export`
+    - :attr:`global`
+
+  * - :cmd:`Definition`, :cmd:`Lemma`, :cmd:`Axiom`, ...
+    - same as :attr:`global`
+    - available with qualified name
+    - attribute not supported
+    - available with short name if imported, with qualified name if not
+
+  * - :cmd:`Notation`
+    - same as :attr:`global`
+    - not available
+    - attribute not supported
+    - available when imported
+
+  * - :cmd:`Notation (abbreviation)`
+    - same as :attr:`global`
+    - not available
+    - attribute not supported
+    - available with short name if imported, with qualified name if not
+
+  * - ``Hints`` (and :cmd:`Instance`)
+    - same as :attr:`export`
+    - not available
+    - available when imported
+    - always available outside
+
+  * - :cmd:`Set` or :cmd:`Unset` a flag
+    - same as :attr:`local`
+    - no effect
+    - in effect when imported
+    - in effect outside
+
+  * - :cmd:`Canonical Structure`
+    - same as :attr:`local`
+
+      or :attr:`global`
+    - available when imported
+    - attribute not supported
+    - available when imported
+
+  * - :cmd:`Coercion`
+    - same as :attr:`global`
+    - not available
+    - attribute not supported
+    - available when imported
+
+  * - :cmd:`Ltac`
+    - same as :attr:`global`
+    - not available
+    - attribute not supported
+    - available with short name if imported, with qualified name if not
```

Comment:
Let's hope this longish text gets line breaks in the rendered doc.  IIRC that may not be the case.  (Still waiting for the doc job to run.)

Reply:
Alas, the lines are not split.  The user would have to scroll the table right/left--much harder to read.  You might be able to fit it in the available width by much shorter mnemonics explained just above or below the main table.  They could be hyperlinked if you like.

![image](https://github.com/coq/coq/assets/1253341/546b64eb-ed2d-421d-b882-763a9cd5d9b4)